File: py/7.py
class Solution(object):
    def reverse(self, x):
        """
        :type x: int
        :rtype: int
        """
        ret = int(str(x)[::-1]) if x>=0 else -int(str(x)[1:][::-1])
        if ret > 0x7fffffff or ret <= -0x80000000:
            return 0
        return int(ret)

File: cpp/7.cpp
class Solution {
 public:
  int reverse(int x) {
    int ret = 0, op = 1;
    if (x < 0) op = -1, x = -x;
    while (x) {
      int t = ret;
      ret = ret * 10 + x % 10;
      if (t != (ret - x % 10) / 10) return 0;
      x /= 10;
    }
    return ret * op;
  }
};
